packages/run-protocol/src/vpool-xyk-amm/addPool.js

```diff
@@ -44,65 +41,54 @@ export const makeAddIssuer = (
       X`${keyword} asset not fungible (must use NAT math)`,
     );
 
-    if (brandToLiquidityIssuer.has(secondaryBrand)) {
-      return brandToLiquidityIssuer.get(secondaryBrand);
-    }
+    /** @type {(brand: Brand) => Promise<ZCFMint>} */
+    const makeLiquidityMint = brand => {
+      assert(
+        !isInSecondaries(brand),
+        X`issuer ${secondaryIssuer} already has a pool`,
+      );
 
-    assert(
-      !isInSecondaries(secondaryBrand),
-      X`issuer ${secondaryIssuer} already has a pool`,
+      const liquidityKeyword = `${keyword}Liquidity`;
+      zcf.assertUniqueKeyword(liquidityKeyword);
+
+      return E.when(
+        Promise.all([
+          zcf.saveIssuer(secondaryIssuer, keyword),
+          zcf.makeZCFMint(liquidityKeyword, AssetKind.NAT, DISPLAY_INFO),
+        ]),
+      )
+        .then(([issuer, mint]) => {
+          console.log(
+            X`Saved issuer ${secondaryIssuer} to keyword ${keyword} and got back ${issuer}`,
+          );
+          return mint;
+        })
+        .catch(e => {
+          console.error(
+            X`Failure Saving issuer ${secondaryIssuer}. Not added to Reserve`,
+          );
+          throw e;
+        });
+    };
+    /** @type {(brand: Brand) => Promise<void>} */
```

Comment:
I'd like to see a blank line between `makeLiquidityMint` and `finish`.  The break was hard for me to parse visually.